sdk/tables/aztable/connection_string.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package aztable
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+var ErrConnectionString = errors.New("connection string is either blank or malformed")
+
+func NewTableClientFromConnectionString(tableName string, connectionString string, options *TableClientOptions) (*TableClient, error) {
+	endpoint, credential, err := parseConnectionString(connectionString)
+	if err != nil {
+		return nil, err
+	}
+	return NewTableClient(tableName, endpoint, *credential, options)
+}
+
+func NewTableServiceClientFrommConnectionString(connectionString string, options *TableClientOptions) (*TableServiceClient, error) {
+	endpoint, credential, err := parseConnectionString(connectionString)
+	if err != nil {
+		return nil, err
+	}
+	return NewTableServiceClient(endpoint, *credential, options)
+}
+
+func convertConnStrToMap(connStr string) (map[string]string, error) {
+	ret := make(map[string]string)
+	connStr = strings.TrimRight(connStr, ";")
+
+	splitString := strings.Split(connStr, ";")
+	if len(splitString) == 0 {
+		return ret, ErrConnectionString
+	}
+	for _, stringPart := range splitString {
+		parts := strings.Split(stringPart, "=")
+		if len(parts) != 2 {
+			return ret, ErrConnectionString
+		}
+		ret[parts[0]] = parts[1]
+	}
+	return ret, nil
+}
+
+func parseConnectionString(connStr string) (string, *azcore.Credential, error) {
+	var serviceURL string
+	var cred azcore.Credential
+
+	connStrMap, err := convertConnStrToMap(connStr)
+	if err != nil {
+		return "", nil, err
+	}
+
+	accountName, ok := connStrMap["AccountName"]
+	if !ok {
+		return "", nil, ErrConnectionString
+	}
+	accountKey, ok := connStrMap["AccountKey"]
+	if !ok {
+		return "", nil, ErrConnectionString
+	}
+
+	if accountName == "" || accountKey == "" {
+		// Try sharedaccesssignature
+		sharedAccessSignature, ok := connStrMap["sharedaccesssignature"]
+		if !ok {
+			return serviceURL, nil, ErrConnectionString
+		}
+		return sharedAccessSignature, nil, errors.New("there is not support for SharedAccessSignature yet")
```

Comment:
```suggestion
		return sharedAccessSignature, nil, errors.New("a SharedAccessSignature are not supported")
```

'yet' is kind of weird when it doesn't have a timeframe. There was a mild typo ('there is not support' should probably be 'there is no support') if you want to stick with the original text.

Reply:
Good point, although this is a personal placeholder and I intend on having SAS complete before the first preview release of tables goes out.